### PR TITLE
Pm 2021 06 second step add image

### PR DESCRIPTION
**Pull request by phillimorland**
Still to do: 

- [x] funtion on addImageButton to make 'camera' and 'lirary' buttons appear when pressed
- [ ] replace deprecated expo permissions package
- [x] update naming 
- [x] reafactor and complete styling, including using variables 
- [x] cropping ratio to be checked and adjusted and image quality to ensure looks good on both mobile and desktop
- [x] move image form fields to seperate file?
- [x] need whole screen to be scrollable
- [x] description form fields char limit
- [x] stop form having capital at start
- [x] make whole checkbox clickable

### CODE REVIEW

**Comment by phillimorland** [comment]
I think the permissions thing can be added to the technical review for next sprint as i want to see how often the other packages we are using get updated, if they are not very stable then we might need a more significant refactor then just swapping it out.

File: components/formFields.styles.js [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 import { COLORS } from '../theme/colors';
+import { SPACINGS } from '../theme/spacings';
+import { SIZES, LINEHEIGHTS } from '../theme/fonts';
 
 export const styles = StyleSheet.create({
   formTitle: {
-    fontSize: 17,
-    lineHeight: 25,
+    fontSize: SIZES.md,
+    lineHeight: LINEHEIGHTS.md,
```
Comment (khamui) [inline components/formFields.styles.js:9]: nice!!!

File: components/imageFormField.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+export const ImagePickerFormField = () => {
+  const [image, setImage] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS !== 'web') {
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== 'granted') {
+          alert('Sorry, we need camera roll permissions to make this work!');
```
Comment (khamui) [inline components/imageFormField.js:60]: what is the difference to the `Alert` above?
could we use the same?

File: components/imageFormField.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  };
+
+  const captureImageHandler = async () => {
+    const hasPermission = await verifyPermissions();
+    if(!hasPermission) {
+      return;
+    }
+    const image = await ImagePicker.launchCameraAsync({
+      allowsEditing: true,
+      aspect: [16, 9],
```
Comment (khamui) [inline components/imageFormField.js:30]: super detail (probably for later): we could extract this to a config variable or file and reuse it also for below (picklibraryimage).

**Review by khamui — APPROVED** [review]

**Comment by khamui** [comment]
@phillimorland very nice state of the image adding feature!! code is very readable. I tested it and it works, except of the scrolling somehow (android). And i have some comments on tiny stuff (that definitely is for later).

But I think this state is good to merge! Let me know if you want to touch it or I can merge it.

**Comment by phillimorland** [comment]
@kha i'll check the alert stuff and tag you again, rest I think we can leave for now, the scroll on android is interesting as works for me locally but I think it's becasue we have a dropdown inside a scrollable area which isn't right, I added an issue for it.

**Comment by Kha** [comment]
> and tag you again

Please don't, because I'm not @khamui

**Comment by fuzzylogic2000** [comment]
@khamui I will also mention you here.

**Comment by fuzzylogic2000** [comment]
@khamui No, you're still not first in the list for PRs. Well, will have to comment more!

File: components/imageFormField.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+export const ImagePickerFormField = () => {
+  const [image, setImage] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS !== 'web') {
+        const { status } = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (status !== 'granted') {
+          Alert.alert('Sorry, we need camera roll permissions to make this work!');
```
Comment (khamui) [inline components/imageFormField.js:61]: nice!

**Pull request merged**